apps/gallery/js/MetadataParser.js

```diff
@@ -500,60 +586,67 @@ var metadataParser = (function() {
         canvas.toBlob(function(blob) {
```

Comment:
Quick clarification: Shouldn't we call  metadataError('createThumbnailAndPreview: Image failed to load') and return; 
instead of error('createThumbnailAndPreview: Image failed to load'); on Line 507